RecoCTPPS/PixelLocal/plugins/CTPPSPixelLocalTrackProducer.cc

```diff
@@ -0,0 +1,258 @@
+  desc.add<double>         ("RPixRoadRadius"                   , 1.0                           );
+  desc.add<int>            ("RPixMinRoadSize"                  , 3                             );
+  desc.add<int>            ("RPixMaxRoadSize"                  , 20                            );
+
+  descriptions.add("ctppsPixelLocalTracks", desc);
+}
+
+//------------------------------------------------------------------------------------------------//
+
+void CTPPSPixelLocalTrackProducer::produce(edm::Event& iEvent, const edm::EventSetup& iSetup){
+  // Step A: get inputs
+  
+  edm::Handle<edm::DetSetVector<CTPPSPixelRecHit> > recHits;
+  iEvent.getByToken(tokenCTPPSPixelRecHit_, recHits);
+  const edm::DetSetVector<CTPPSPixelRecHit> &recHitVectorOrig = *recHits;
+  edm::DetSetVector<CTPPSPixelRecHit> recHitVector = recHitVectorOrig;
+
+  // get geometry
+  edm::ESHandle<CTPPSGeometry> geometryHandler;
+  iSetup.get<VeryForwardRealGeometryRecord>().get(geometryHandler);
+  const CTPPSGeometry &geometry = *geometryHandler;
+  geometryWatcher_.check(iSetup);
+
+  std::vector<CTPPSPixelDetId> listOfPotWithHighOccupancyPlanes; 
+  std::map<CTPPSPixelDetId, uint32_t> mapHitPerPot;
+
+
+  std::map<CTPPSPixelDetId, TMatrixD> planeRotationMatrixMap;
+  std::map<CTPPSPixelDetId, CLHEP::Hep3Vector> planePointMap;
+
+  for(const auto & recHitSet : recHitVector){
+    
+    if(verbosity_>2) std::cout<<"Hits found in plane = "<<recHitSet.detId()<< " number = " << recHitSet.size() <<std::endl;
+    CTPPSPixelDetId tmpRomanPotId = CTPPSPixelDetId(recHitSet.detId()).getRPId();
+    uint32_t hitOnPlane = recHitSet.size();
+
+    //Get the number of hits per pot
+    if(mapHitPerPot.find(tmpRomanPotId)==mapHitPerPot.end()){
+      mapHitPerPot[tmpRomanPotId] = hitOnPlane;
+
+      //I store a point on the plane and the rotation matrix for all the planes on the pot
+      for(const auto & plane : listOfAllPlanes_){
+        CTPPSPixelDetId tmpDetectorId = tmpRomanPotId;
+        tmpDetectorId.setPlane(plane);
+        DDRotationMatrix theRotationMatrix = geometry.getSensor(tmpDetectorId)->rotation();
+        TMatrixD theRotationTMatrix;
+        theRotationTMatrix.ResizeTo(3,3);
+        theRotationMatrix.GetComponents(theRotationTMatrix[0][0], theRotationTMatrix[0][1], theRotationTMatrix[0][2],
+                                        theRotationTMatrix[1][0], theRotationTMatrix[1][1], theRotationTMatrix[1][2],
+                                        theRotationTMatrix[2][0], theRotationTMatrix[2][1], theRotationTMatrix[2][2]);
+        planeRotationMatrixMap[tmpDetectorId] = TMatrixD();
+        planeRotationMatrixMap[tmpDetectorId].ResizeTo(3,3);
+        planeRotationMatrixMap[tmpDetectorId] = theRotationTMatrix;
+        CLHEP::Hep3Vector localV(0.,0.,0.);
+        CLHEP::Hep3Vector globalV = geometry.localToGlobal(tmpDetectorId,localV);
+        planePointMap[tmpDetectorId] = globalV;
```

Comment:
all of these transformations should be available from the geometry interface.

Reply:
Dear Slava,
the idea to temporary store the rotation matrix and a point on the detector plane was to avoid calling again the geometry in the RPixDetTrackFinder. We though that this would be a more efficient way to run. Do you think that it is better to use the geometry methods?